### ISSUE TITLE
refactor torch_dataset folder

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/ocf-data-sampler#78]
Should only torch dataset files be in the torch_dataset folder. 
We could move `process_and_combine` out of there

**Comment by Sukh-P** [comment]
> Should only torch dataset files be in the torch_dataset folder.

Yep I think so, feels easy to understand that folder then, or as I mentioned in the other PR at the very least there be another folder within `torch_dataset` which holds all non torch Dataset files so it's a bit clearer

**Comment by RupeshMangalam21** [comment]
Hi, just wanted to know if we still need this refactoring? I assume we do, since there are currently two files in `torch_dataset` that are unrelated, `merge_and_fill_utils.py` and `valid_time_periods.py` in `torch_dataset` folder.
